src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import command.*;
+
+/**
+ * Parser reads the primary user input and deals with making sense of the input.
+ */
+public class Parser {
+    private static TaskList taskList;
+    private static Ui ui;
+
+    /**
+     * The constructor of Parser.
+     *
+     * @param taskList The task list to be passed into the Command.
+     * @param ui The ui of the chatbot to be passed into the Command.
+     */
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Returns a specific Command when the corresponding command keywords are
+     * detected in the user input.
+     *
+     * @param fullCommand The full input the user typed into the chatbot.
+     * @return A specific Command that corresponds to the command keyword the user inputs.
+     */
+    public static Command parse(String fullCommand) {
+        String command = fullCommand.split(" ")[0];
+        Keyword k = Keyword.valueOf(command.toUpperCase());
+        switch (k) {
```

Comment:
I like how you followed the convention for switch statement. Looks like you have modified the format settings in your IDE 👍